domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotter.kt

```diff
@@ -0,0 +1,209 @@
+            previousSnapshot =
+              performanceMetricsAssessor.computeSnapshotAtCurrentTime(currentIconification)
+            sendScheduleTakeSnapshotCommand(currentIconification, switchIconificationCount)
+          }
+          is CommandMessage.ScheduleTakeSnapshot -> scheduleTakeSnapshot(
+            message.currentIconification,
+            switchIconificationCount
+          )
+          is CommandMessage.TakeSnapshot -> {
+            if (message.switchId == switchIconificationCount) {
+              val newSnapshot =
+                performanceMetricsAssessor.computeSnapshotAtCurrentTime(currentIconification)
+              sendLogSnapshotDiffCommand(
+                performanceMetricsAssessor.getRelativeCpuUsage(previousSnapshot, newSnapshot)
+              )
+              previousSnapshot = newSnapshot
+              sendScheduleTakeSnapshotCommand(currentIconification, switchIconificationCount)
+            }
+          }
+          is CommandMessage.LogSnapshotDiff -> {
+            performanceMetricsLogger.logCpuUsage(
+              currentIconification.toScreenName(),
+              message.relativeCpuUsage
+            )
+          }
+        }
+      }
+    }.also {
+      CoroutineScope(backgroundCoroutineDispatcher).launch {
+        sendTakeSnapshotCommand(switchIconificationCount)
+      }
+    }
+  }
+
+  private fun sendSwitchIconificationCommand(newIconification: AppIconification) {
+    commandQueue.offer(CommandMessage.SwitchIconification(newIconification)).apply {
+      if (!this) {
+        val exception = IllegalStateException("Iconification switching failed")
+        consoleLogger.e(
+          "CpuPerformanceSnapshotter",
+          "Failure while switching AppIconification.",
+          exception
+        )
+        exceptionLogger.logException(exception)
+      }
+    }
+  }
+
+  private suspend fun sendScheduleTakeSnapshotCommand(
+    currentIconification: AppIconification,
+    switchId: Int
+  ) { commandQueue.send(CommandMessage.ScheduleTakeSnapshot(currentIconification, switchId)) }
+
+  private suspend fun sendTakeSnapshotCommand(switchId: Int) {
+    commandQueue.send(CommandMessage.TakeSnapshot(switchId))
+  }
+
+  private suspend fun sendLogSnapshotDiffCommand(relativeCpuUsage: Double) {
+    commandQueue.send(CommandMessage.LogSnapshotDiff(relativeCpuUsage))
+  }
+
+  /**
+   * Schedules a delay on the basis of [currentIconification] and then sends the [CommandMessage]
+   * for taking a [Snapshot] that'll be used for logging the relative cpu-usage of the application.
+   */
+  private fun scheduleTakeSnapshot(currentIconification: AppIconification, switchId: Int) {
+    CoroutineScope(backgroundCoroutineDispatcher).launch {
+      delay(currentIconification.getDelay())
+      sendTakeSnapshotCommand(switchId)
+    }.invokeOnCompletion {
+      it?.let {
+        consoleLogger.e(
+          "PerformanceMetricsController",
+          "Failed to remove metric log.",
```

Comment:
Are these the correct tag and message values to be passing to the logger in this context?

Reply:
oops, modified now.